core/web/api/observable.py

```diff
@@ -49,9 +49,9 @@ def new(self):
             obs = self.objectmanager.objects.get(id=params.pop("id"))
         else:
             if params.pop('refang', None):
-                obs = self.objectmanager.add_text(refang(params.pop('value')))
+                obs = self.objectmanager.add_text(refang(params.pop('value')), type=params.pop('type', None))
             else:
-                obs = self.objectmanager.add_text(params.pop('value'))
+                obs = self.objectmanager.add_text(params.pop('value'), type=params.pop('type', None))
```

Comment:
This will break if `type` is not within the types registered in Yeti.

I would raise an exception in `instantiate_subclass` and catch it here, returning an error (40X) to the API client